failure with chdir

[//]: # (
. Note: for support questions, please use Stackoverflow or Gitter**. 
. This repository's issues are reserved for feature requests and bug reports.
.
. In case of any problems with Allure Jenkins plugin** please use the following repository 
. to create an issue: https://github.com/jenkinsci/allure-plugin/issues
.
. Make sure you have a clear name for your issue. The name should start with a capital 
. letter and no dot is required in the end of the sentence. An example of good issue names:
.
. - The report is broken in IE11
. - Add an ability to disable default plugins
. - Support emoji in test descriptions
)

#### I'm submitting a ... 
  - [x] bug report
  - [ ] feature request
  - [ ] support request => Please do not submit support request here, see note at the top of this template.

#### What is the current behavior?

`pytest-allure` fails and/or crashes when a test case changes directories.


#### If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem

poc.py:
```python
import os
import pytest


@pytest.fixture
def workdir(tmpdir):
    start_dir = os.getcwd()
    os.chdir(tmpdir)
    yield tmpdir
    os.chdir(start_dir)

# this test will generate an erroneous failure report
# for an unknown reason, the capsys fixture presence is necessary to reproduce
def test_allure(workdir, capsys):
    assert True

# this test will generate a stacktrace but no failure
def test_allure2(tmpdir):
    os.chdir(tmpdir)
    assert True
```

`pytest poc.py`


#### What is the expected behavior?

A passing test suite with no stacktrace.

#### What is the motivation / use case for changing the behavior?

This is a common test pattern and the primary use case of the `tmpdir` standard fixture.

#### Please tell us about your environment:

- Allure version:        2.9.0
- Test framework:        pytest@4.1.1
- Allure adaptor:        allure-pytest@2.5.5

#### Other information

test_allure stacktrace:
```
platform linux -- Python 3.6.6, pytest-4.1.1, py-1.7.0, pluggy-0.8.1 -- /bin/python
cachedir: .pytest_cache
rootdir: /tmp, inifile:
plugins: pylint-0.12.3, pudb-0.7.0, mock-1.10.0, cov-2.6.1, pyfakefs-3.5.6, allure-pytest-2.5.5
collected 1 item

poc.py::test_allure PASSED
poc.py::test_allure ERROR

======================================================= ERRORS ========================================================
__________________________________________ ERROR at teardown of test_allure ___________________________________________

tp = <class 'FileNotFoundError'>, value = None, tb = None

    def reraise(tp, value, tb=None):
        try:
            if value is None:
                value = tp()
            if value.__traceback__ is not tb:
                raise value.with_traceback(tb)
>           raise value

/lib/python3.6/site-packages/six.py:693:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/lib/python3.6/site-packages/six.py:693: in reraise
    raise value
/lib/python3.6/site-packages/allure_pytest/listener.py:147: in pytest_fixture_post_finalizer
    self.allure_logger.stop_group(container_uuid, stop=now())
/lib/python3.6/site-packages/allure_commons/reporter.py:46: in stop_group
    plugin_manager.hook.report_container(container=group)
/lib/python3.6/site-packages/allure_commons/logger.py:43: in report_container
    self._report_item(container)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <allure_commons.logger.AllureFileLogger object at 0x7f9560680748>
item = TestResultContainer(uuid='d4daf6b7-9e1f-4dad-bbb6-05c1a3b5521d', name=None, children=['9183410e-6ca7-4ac9-84f3-e41aa8c...hments=[], parameters=[], start=1549034208516, stop=1549034208516)], links=[], start=1549034208513, stop=1549034208516)

    def _report_item(self, item):
        indent = INDENT if os.environ.get("ALLURE_INDENT_OUTPUT") else None
        filename = item.file_pattern.format(prefix=uuid.uuid4())
        data = asdict(item, filter=lambda attr, value: not (type(value) != bool and not bool(value)))
>       with io.open(os.path.join(self._report_dir, filename), 'w', encoding='utf8') as json_file:
E       FileNotFoundError: [Errno 2] No such file or directory: 'allure/7d8f90ab-4591-4de0-8ca1-19f5a3d0d2e5-container.json'

/lib/python3.6/site-packages/allure_commons/logger.py:31: FileNotFoundError
```

test_allure2 stacktrace:
```
platform linux -- Python 3.6.6, pytest-4.1.1, py-1.7.0, pluggy-0.8.1 -- /bin/python
cachedir: .pytest_cache
rootdir: /tmp, inifile:
plugins: pylint-0.12.3, pudb-0.7.0, mock-1.10.0, cov-2.6.1, pyfakefs-3.5.6, allure-pytest-2.5.5
collected 1 item

poc.py::test_allure2 PASSED
INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/lib/python3.6/site-packages/_pytest/main.py", line 203, in wrap_session
INTERNALERROR>     session.exitstatus = doit(config, session) or 0
INTERNALERROR>   File "/lib/python3.6/site-packages/_pytest/main.py", line 243, in _main
INTERNALERROR>     config.hook.pytest_runtestloop(session=session)
INTERNALERROR>   File "/lib/python3.6/site-packages/pluggy/hooks.py", line 284, in __call__
INTERNALERROR>     return self._hookexec(self, self.get_hookimpls(), kwargs)
INTERNALERROR>   File "/lib/python3.6/site-packages/pluggy/manager.py", line 68, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "/lib/python3.6/site-packages/pluggy/manager.py", line 62, in <lambda>
INTERNALERROR>     firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
INTERNALERROR>   File "/lib/python3.6/site-packages/pluggy/callers.py", line 208, in _multicall
INTERNALERROR>     return outcome.get_result()
INTERNALERROR>   File "/lib/python3.6/site-packages/pluggy/callers.py", line 80, in get_result
INTERNALERROR>     raise ex[1].with_traceback(ex[2])
INTERNALERROR>   File "/lib/python3.6/site-packages/pluggy/callers.py", line 187, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/lib/python3.6/site-packages/_pytest/main.py", line 264, in pytest_runtestloop
INTERNALERROR>     item.config.hook.pytest_runtest_protocol(item=item, nextitem=nextitem)
INTERNALERROR>   File "/lib/python3.6/site-packages/pluggy/hooks.py", line 284, in __call__
INTERNALERROR>     return self._hookexec(self, self.get_hookimpls(), kwargs)
INTERNALERROR>   File "/lib/python3.6/site-packages/pluggy/manager.py", line 68, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "/lib/python3.6/site-packages/pluggy/manager.py", line 62, in <lambda>
INTERNALERROR>     firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
INTERNALERROR>   File "/lib/python3.6/site-packages/pluggy/callers.py", line 208, in _multicall
INTERNALERROR>     return outcome.get_result()
INTERNALERROR>   File "/lib/python3.6/site-packages/pluggy/callers.py", line 80, in get_result
INTERNALERROR>     raise ex[1].with_traceback(ex[2])
INTERNALERROR>   File "/lib/python3.6/site-packages/pluggy/callers.py", line 187, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/lib/python3.6/site-packages/_pytest/runner.py", line 78, in pytest_runtest_protocol
INTERNALERROR>     runtestprotocol(item, nextitem=nextitem)
INTERNALERROR>   File "/lib/python3.6/site-packages/_pytest/runner.py", line 94, in runtestprotocol
INTERNALERROR>     reports.append(call_and_report(item, "teardown", log, nextitem=nextitem))
INTERNALERROR>   File "/lib/python3.6/site-packages/_pytest/runner.py", line 175, in call_and_report
INTERNALERROR>     report = hook.pytest_runtest_makereport(item=item, call=call)
INTERNALERROR>   File "/lib/python3.6/site-packages/pluggy/hooks.py", line 284, in __call__
INTERNALERROR>     return self._hookexec(self, self.get_hookimpls(), kwargs)
INTERNALERROR>   File "/lib/python3.6/site-packages/pluggy/manager.py", line 68, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "/lib/python3.6/site-packages/pluggy/manager.py", line 62, in <lambda>
INTERNALERROR>     firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
INTERNALERROR>   File "/lib/python3.6/site-packages/pluggy/callers.py", line 203, in _multicall
INTERNALERROR>     gen.send(outcome)
INTERNALERROR>   File "/lib/python3.6/site-packages/allure_pytest/listener.py", line 189, in pytest_runtest_makereport
INTERNALERROR>     self.attach_data(report.caplog, "log", AttachmentType.TEXT, None)
INTERNALERROR>   File "/lib/python3.6/site-packages/allure_pytest/listener.py", line 198, in attach_data
INTERNALERROR>     self.allure_logger.attach_data(uuid4(), body, name=name, attachment_type=attachment_type, extension=extension)
INTERNALERROR>   File "/lib/python3.6/site-packages/allure_commons/reporter.py", line 117, in attach_data
INTERNALERROR>     plugin_manager.hook.report_attached_data(body=body, file_name=file_name)
INTERNALERROR>   File "/lib/python3.6/site-packages/pluggy/hooks.py", line 284, in __call__
INTERNALERROR>     return self._hookexec(self, self.get_hookimpls(), kwargs)
INTERNALERROR>   File "/lib/python3.6/site-packages/pluggy/manager.py", line 68, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "/lib/python3.6/site-packages/pluggy/manager.py", line 62, in <lambda>
INTERNALERROR>     firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
INTERNALERROR>   File "/lib/python3.6/site-packages/pluggy/callers.py", line 208, in _multicall
INTERNALERROR>     return outcome.get_result()
INTERNALERROR>   File "/lib/python3.6/site-packages/pluggy/callers.py", line 80, in get_result
INTERNALERROR>     raise ex[1].with_traceback(ex[2])
INTERNALERROR>   File "/lib/python3.6/site-packages/pluggy/callers.py", line 187, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/lib/python3.6/site-packages/allure_commons/logger.py", line 53, in report_attached_data
INTERNALERROR>     with open(destination, 'wb') as attached_file:
INTERNALERROR> FileNotFoundError: [Errno 2] No such file or directory: 'allure/1c6d698d-e01c-4a7a-9af1-b6ed5560588f-attachment.txt'
```

[//]: # (
. e.g. detailed explanation, stacktraces, related issues, suggestions 
. how to fix, links for us to have more context, eg. Stackoverflow, Gitter etc
)

--- a/allure-pytest/src/plugin.py
+++ b/allure-pytest/src/plugin.py
@@ -2,6 +2,7 @@ import argparse
 
 import allure
 import allure_commons
+import os
 
 from allure_commons.types import LabelType
 from allure_commons.logger import AllureFileLogger
@@ -115,6 +116,7 @@ def pytest_configure(config):
     config.add_cleanup(cleanup_factory(test_helper))
 
     if report_dir:
+        report_dir = os.path.abspath(report_dir)
         test_listener = AllureListener(config)
         config.pluginmanager.register(test_listener)
         allure_commons.plugin_manager.register(test_listener)

